Merchant/App/PaymentInput/PaymentInputViewController.swift

```diff
@@ -150,6 +173,13 @@ extension PaymentInputViewController: PinViewControllerDelegate {
             }
             presenter?.didSetAmount(amountStr)
         case "0", "1", "2", "3", "4", "5", "6", "7", "8", "9":
+            let range = amountStr.range(of: ".")
+            if let r = range {
+                let decimals = amountStr[r.upperBound..<amountStr.endIndex]
```

Comment:
you can do the if-let condition for the range check here.. if let range = amountStr.range(of: ".") --- so you don't need to check again it again, and remove if let r = range